pennylane/ops/qutrit/non_parametric_ops.py

```diff
@@ -202,3 +203,200 @@ def adjoint(self):
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int]): the wires the operation acts on
+    """
+    num_wires = 2
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    @staticmethod
+    def compute_matrix():
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.TAdd.matrix`
+
+        Returns:
+            ndarray: matrix
+
+        **Example**
+
+        >>> print(qml.TAdd.compute_matrix())
+        [[1 0 0 0 0 0 0 0 0]
+         [0 1 0 0 0 0 0 0 0]
+         [0 0 1 0 0 0 0 0 0]
+         [0 0 0 0 0 1 0 0 0]
+         [0 0 0 1 0 0 0 0 0]
+         [0 0 0 0 1 0 0 0 0]
+         [0 0 0 0 0 0 0 1 0]
+         [0 0 0 0 0 0 0 0 1]
+         [0 0 0 0 0 0 1 0 0]]
+        """
+        return np.array(
+            [
+                [1, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 1, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 1, 0, 0, 0],
+                [0, 0, 0, 1, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 1, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 1, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 1],
+                [0, 0, 0, 0, 0, 0, 1, 0, 0],
+            ]
+        )
+
+    @staticmethod
+    def compute_eigvals():
+        r"""Eigenvalues of the operator in the computational basis (static method).
+
+        If :attr:`diagonalizing_gates` are specified and implement a unitary :math:`U`,
+        the operator can be reconstructed as
+
+        .. math:: O = U \Sigma U^{\dagger},
+
+        where :math:`\Sigma` is the diagonal matrix containing the eigenvalues.
+        Otherwise, no particular order for the eigenvalues is guaranteed.
+
+        .. seealso:: :meth:`~.TAdd.eigvals`
+
+        Returns:
+            array: eigenvalues
+
+        **Example**
+
+        >>> print(qml.TAdd.compute_eigvals())
+        [-0.5+0.8660254j -0.5-0.8660254j  1. +0.j        -0.5+0.8660254j -0.5-0.8660254j  1. +0.j         1. +0.j         1. +0.j         1. +0.j       ]
+        """
+        return np.array([OMEGA, OMEGA**2, 1, OMEGA, OMEGA**2, 1, 1, 1, 1])
+
+    # TODO: Add compute_decomposition() once parametric ops are added.
+
+    def pow(self, z):
+        if isinstance(z, int):
+            z_mod3 = z % 3
+            if z_mod3 < 2:
+                return super().pow(z_mod3)
+            return [self.adjoint()]
+        return super().pow(z)
+
+    def adjoint(self):
+        op = TAdd(self.wires)
+        op.inverse = not self.inverse
+        return op
+
+    @property
```

Comment:
Now that this is included, we could also set the `_controlled` attribute of the `TShift` operation to be `TAdd`, (as is done with `PauliX` and `CNOT` in the qubit case). This could be done, though, in a separate, smaller PR as it may require a number of new tests, and also may need to be done for other operations.

Reply:
This can't be done, as `TShift` always shifts by one index, while the number of times `TAdd` shifts the target qutrit depends on the state of the control qutrit.